src/giskardpy/goals/align_to_push_door.py

```diff
@@ -0,0 +1,125 @@
+                 end_condition: cas.Expression = cas.TrueSymbol):
+        """
+        The objective is to reach an intermediate point before pushing the door
+        """
+        self.root = god_map.world.search_for_link_name(root_link, root_group)
+        self.tip = god_map.world.search_for_link_name(tip_link, tip_group)
+        self.door_object = god_map.world.search_for_link_name(door_object)
+        self.object_joint_angle = \
+            god_map.world.state.to_position_dict()[god_map.world.search_for_joint_name(object_joint_name)]
+
+        tip_gripper_axis.header.frame_id = self.tip
+        tip_gripper_axis.vector = tf.normalize(tip_gripper_axis.vector)
+        object_rotation_axis.header.frame_id = self.door_object
+        object_rotation_axis.vector = tf.normalize(object_rotation_axis.vector)
+
+        self.tip_gripper_axis = tip_gripper_axis
+        self.object_rotation_axis = object_rotation_axis
+        self.root_P_door_object = PointStamped()
+        self.root_P_door_object.header.frame_id = self.root
+
+        # Pose of the articulated object is at the center of the axis along which it rotates and not at the center of
+        # the object.
+        self.door_length = door_length
+        self.door_height = door_height
+        self.reference_linear_velocity = reference_linear_velocity
+        self.reference_angular_velocity = reference_angular_velocity
+        self.weight = weight
+        if name is None:
+            name = f'{self.__class__.__name__}/{self.root}/{self.tip}'
+        super().__init__(name=name)
+
+        root_T_tip = god_map.world.compose_fk_expression(self.root, self.tip)
+        root_T_door_expr = god_map.world.compose_fk_expression(self.root, self.door_object)
+        object_V_object_rotation_axis = cas.Vector3(self.object_rotation_axis)
+        tip_V_tip_grasp_axis = cas.Vector3(self.tip_gripper_axis)
+        root_V_object_rotation_axis = cas.dot(root_T_door_expr, object_V_object_rotation_axis)
+        root_V_tip_grasp_axis = cas.dot(root_T_tip, tip_V_tip_grasp_axis)
+        root_T_door = god_map.world.compute_fk_pose(self.root, self.door_object)
+        door_T_root = god_map.world.compute_fk_pose(self.door_object, self.root)
+
+        root_P_intermediate_point = PointStamped()
+        root_P_intermediate_point.header.frame_id = self.root
+        root_P_intermediate_point.point.x = root_T_door.pose.position.x
+        root_P_intermediate_point.point.y = root_T_door.pose.position.y
+        root_P_intermediate_point.point.z = root_T_door.pose.position.z
+        if root_V_object_rotation_axis.y == 1:
+            # 3/4 of the height as the tip has to be a little further inside the object
+            root_P_intermediate_point.point.z = root_P_intermediate_point.point.z + self.door_height*3/4
+        elif root_V_object_rotation_axis.z == 1:
+            root_P_intermediate_point.point.y = root_P_intermediate_point.point.y + self.door_length*3/4
```

Comment:
this assumes that the axis of all doors are aligned the same way?

Reply:
No. but assumes the door is in yz plane as x would be along object depth